File: ui/log_panel.py
# ui/log_panel.py
from PyQt6.QtWidgets import QTextEdit
from PyQt6.QtGui import QTextCursor
from datetime import datetime


class LogPanel(QTextEdit):
    COLOR_MAP = {
        "INFO": "#4CAF50",
        "WARNING": "#FF9800",
        "ERROR": "#F44336",
        "DEBUG": "#9E9E9E",
    }

    def __init__(self, parent=None):
        super().__init__(parent)
        self.setReadOnly(True)
        self.setMaximumHeight(160)
        self.setStyleSheet("""
            QTextEdit {
                background: #1e1e1e;
                color: #d4d4d4;
                font-family: 'Consolas', 'Courier New', monospace;
                font-size: 12px;
                border: 1px solid #3c3c3c;
                border-radius: 4px;
            }
        """)

    def append_log(self, level: str, message: str):
        color = self.COLOR_MAP.get(level.upper(), "#d4d4d4")
        ts = datetime.now().strftime("%H:%M:%S")
        html = (
            f'<span style="color: #808080;">[{ts}]</span> '
            f'<span style="color: {color};">{level.upper()}</span> '
            f'<span style="color: #d4d4d4;">{message}</span>'
        )
        self.append(html)
        self.moveCursor(QTextCursor.MoveOperation.End)

    def append_info(self, message: str):
        self.append_log("INFO", message)

    def append_warning(self, message: str):
        self.append_log("WARNING", message)

    def append_error(self, message: str):
        self.append_log("ERROR", message)

    def append_debug(self, message: str):
        self.append_log("DEBUG", message)

    def clear_logs(self):
        self.clear()

File: src/log_panel.cpp
#include "log_panel.h"
#include <QDateTime>
#include <QTextCharFormat>
#include <QDebug>

LogPanel::LogPanel(QWidget* parent)
    : QTextEdit(parent)
{
    setReadOnly(true);
    setStyleSheet(R"(
        QTextEdit {
            background: #F8FAFC;
            color: #475569;
            font-family: 'Cascadia Code', 'Consolas', 'Courier New', monospace;
            font-size: 11px;
            border: none;
            padding: 4px 2px;
        }
    )");
}

QString LogPanel::ts() const {
    return QDateTime::currentDateTime().toString("HH:mm:ss");
}

void LogPanel::log(const QString& level, const QString& msg, const QString& color) {
    QString html = QString(
        "<span style=\"color: #94A3B8;\">[%1]</span> "
        "<span style=\"color: %2;\">%3</span> "
        "<span style=\"color: #475569;\">%4</span>"
    ).arg(ts()).arg(color).arg(level.toUpper()).arg(msg);
    append(html);
}

void LogPanel::appendInfo(const QString& msg)    { log("INFO", msg, "#3B82F6"); }
void LogPanel::appendWarning(const QString& msg) { log("WARN", msg, "#F59E0B"); }
void LogPanel::appendError(const QString& msg)  { log("ERRO", msg, "#EF4444"); }
void LogPanel::appendDebug(const QString& msg)   { log("DEBU", msg, "#94A3B8"); }
